src/frontend/packages/git/src/shared/scm/scm-base.ts

```diff
@@ -21,25 +21,29 @@ export abstract class BaseSCM {
 
   constructor(public publicApiUrl: string) { }
 
+  public setPublicApi(url: string) {
+    this.publicApiUrl = url
+  }
+
   public getPublicApi(): string {
     return this.publicApiUrl;
   }
 
-  public getAPI(): Observable<GitApiRequest> {
+  public getAPI(options?: HttpOptions): Observable<GitApiRequest> {
```

Comment:
you can also use default values here, then you avoid some unwanted scenarios 
```suggestion
  public getAPI(options = new HttpOptions()): Observable<GitApiRequest> {
```

Reply:
Done 👍🏻